File: week5/SimpleInheritance.cpp
#include<iostream>
using namespace std;

class Lion
{
    public:
        void displayBreed()
        {
            cout<<"I belong to the carnivore type.\nI belong to genus Panthera\n";
        }
        void eat()
        {
            cout<<"I eat meat and cheese\n";
        }

};

class Cub: public Lion
{
    public:
        void play()
        {
            cout<<"I play a lot\nIt helps me to bond to my family\n";
        }
        void skills()
        {
            cout<<"With this I develop lot of skills that I could use later\n";
        }
};

int main()
{
    Cub obj;
    obj.displayBreed();
    obj.eat();
    obj.play();
    obj.skills();
    return 0;
}

File: week5/SimpleInheritance.java
class Lion
{
    public
        void displayBreed()
        {
            System.out.print("I belong to carnivore type\nI belong to genus Panthera\n");
        }
        void eat()
        {
            System.out.print("I eat cheese and meat\n");
        }
}

class Cub extends Lion
{
    public
        void play()
        {
            System.out.print("I play a lot\nIt helps to bond with my family\n");
        }
        void skills()
        {
            System.out.print("With this I develop lot of skills that I could use later\n");
        }
}

public class SimpleInheritance
{
    public static void main(String[] args)
    {
        Cub baby=new Cub();
        baby.displayBreed();;
        baby.eat();
        baby.play();
        baby.skills();
    }
}
